protocols/relay/src/protocol/inbound_hop.rs

```diff
@@ -66,7 +66,7 @@ impl upgrade::InboundUpgrade<NegotiatedSubstream> for Upgrade {
                 .ok_or(FatalUpgradeError::StreamClosed)??;
 
             let r#type =
-                hop_message::Type::from_i32(r#type).ok_or(FatalUpgradeError::ParseTypeField)?;
+                hop_message::Type::from_i32(r#type.unwrap()).ok_or(FatalUpgradeError::ParseTypeField)?;
```

Comment:
```suggestion
                hop_message::Type::from_i32(r#type.ok_or(FatalUpgradeError::MissingType)?).ok_or(FatalUpgradeError::ParseTypeField)?;
```
This will fail without panicking.